presto-main/src/main/java/com/facebook/presto/sql/planner/StatsRecorder.java

```diff
@@ -1,79 +0,0 @@
-/*
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package com.facebook.presto.sql.planner;
-
-import com.facebook.presto.sql.planner.iterative.IterativeOptimizer;
-import com.facebook.presto.sql.planner.iterative.Rule;
-import com.facebook.presto.sql.planner.iterative.RuleStats;
-import org.weakref.jmx.MBeanExporter;
-import org.weakref.jmx.ObjectNames;
-
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.Map;
-
-import static com.google.common.base.Preconditions.checkArgument;
-import static com.google.common.base.Verify.verify;
-import static java.lang.String.format;
-
-public class StatsRecorder
-{
-    private final Map<Class<?>, RuleStats> stats = new HashMap<>();
```

Comment:
> Rename StatsRecorder to RuleStatsRecorder

doesn't quite look like a rename?

also, could you mention in the commit message why you rename the class?

Reply:
also, why we don't simply use same stats objects for rules and optimizers?
If we want to distinguish them on the JMX level, we can do this with names. Now we have stats exported and stats recorder classes duplicated.